tests/test_depsolver.py

```diff
@@ -538,7 +515,50 @@ def _prepare_test_data(pulp):
         requires=[],
     )
 
-    repo_1_units = [unit_1, unit_2, unit_5, unit_11a, unit_11b, unit_11c, unit_12]
+    unit_14 = RpmUnit(
+        name="has-files",
+        version="100",
+        release="200",
+        epoch="1",
+        arch="x86_64",
+        provides=[],
+        requires=[],
+        files=["/some/script"],
+    )
+
+    md_unit_1 = ModulemdUnit(
+        name="test",
+        stream="10",
+        version=100,
+        context="abcdef",
+        arch="x86_64",
+        artifacts=[
+            "perl-version-7:0.99.24-441.module+el8.3.0+6718+7f269185.src",
+            "perl-version-7:0.99.24-441.module+el8.3.0+6718+7f269185.x86_64",
+        ],
+    )
+    md_unit_2 = ModulemdUnit(
+        name="test",
+        stream="20",
+        version=100,
+        context="abcdef",
+        arch="x86_64",
+        artifacts=[
+            "perl-version-7:1.99.24-441.module+el8.4.0+9911+7f269185.src",
+            "perl-version-7:1.99.24-441.module+el8.4.0+9911+7f269185.x86_64",
+        ],
+    )
+
+    repo_1_units = [
+        unit_1,
+        unit_2,
+        unit_5,
+        unit_11a,
+        unit_11b,
+        unit_11c,
+        unit_12,
+        unit_14,
+    ]
```

Comment:
It's not obvious but, by adding unit_14 to the list, it's then expected in the output of the run. The absence of an assertion error is proof that the file dep is resolved.